orttraining/orttraining/core/graph/optimizer/adam_optimizer_builder.cc

```diff
@@ -79,20 +115,36 @@ Status AdamOptimizerBuilder::Build(
         weight_dims.push_back(dim.dim_value());
       }
 
+      auto element_type = ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_FLOAT;
```

Comment:
may we enforce constness by using a ternary operator here?
```
const auto element_type = opt_configs[i].use_mixed_precision_moments ? ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_FLOAT16 : ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_FLOAT
```

unless there are more values that ```element_type``` can take (other than these 2).